packages/next-frontend/src/components/create-campaign/form/dex-step/index.tsx

```diff
@@ -0,0 +1,113 @@
+import { useMemo, type MouseEvent } from "react";
+import { useTranslations } from "next-intl";
+import classNames from "classnames";
+import { Step } from "@/src/components/step";
+import { StepPreview } from "@/src/components/step/preview";
+import { StepContent } from "@/src/components/step/content";
+import { useAvailableAmms } from "@/src/hooks/useAvailableAmms";
+import { animated, useSpring } from "@react-spring/web";
+import { Typography } from "@/src/ui/typography";
+import type { CreateCampaignFormProps } from "..";
+
+import styles from "./styles.module.css";
+
+interface AmmStepProps extends CreateCampaignFormProps {}
+
+export function AmmStep({
```

Comment:
Not a huge fan of how the picking is implemented here. A couple things:

- Why aren't we using `useCallback` for the `handleDexOnClick` function?
- I think we can rename `handleDexOnClick` to `getDexChangeHandler` that takes as an input the dex struct and returns a function that updates the state with that dex struct when invoked.

Also, do you think each step should only update what it wants without having full access to the full `payload`? In this case we'd have a `dex` and `onDexChange` props passed in instead of `payload`, `payloadIndex`, `onPayloadChange`.

Also not clear what `payloadIndex` is at all.

Reply:
So, `payloadIndex` was kind of a leftover (for multiple campaigns, but we'll implement that logic once we have established the proper multi-campaigns workflow); as for the other suggestions I agree, I've implemented the changes here b4a02f6e10f0e8d44a66ac5bd8910584dfe3afb4.